skimage/transform/_warps.py

```diff
@@ -1093,3 +1131,167 @@ def warp_polar(image, center=None, *, radius=None, output_shape=None,
                   output_shape=output_shape, **kwargs)
 
     return warped
+
+
+def _local_mean_weights(old_size, new_size, grid_mode, dtype):
+    """Create a 2D weight matrix for resizing with the local mean.
+
+    Parameters
+    ----------
+    old_size: int
+        Old size.
+    new_size: int
+        New size.
+    grid_mode : bool
+        Whether to use grid data model of pixel/voxel model for
+        average weights computation.
+    dtype: dtype
+        Output array data type.
+
+    Returns
+    -------
+    weights: (new_size, old_size) array
+        Rows sum to 1.
+
+    """
+    if grid_mode:
+        old_breaks = np.linspace(0, old_size, num=old_size + 1, dtype=dtype)
+        new_breaks = np.linspace(0, old_size, num=new_size + 1, dtype=dtype)
+    else:
+        old, new = old_size - 1, new_size - 1
+        old_breaks = np.pad(np.linspace(0.5, old - 0.5, old, dtype=dtype),
+                            1, 'constant', constant_values=(0, old))
+        if new == 0:
+            val = np.inf
+        else:
+            val = 0.5 * old / new
+        new_breaks = np.pad(np.linspace(val, old - val, new, dtype=dtype),
+                            1, 'constant', constant_values=(0, old))
+
+    upper = np.minimum(new_breaks[1:, np.newaxis], old_breaks[np.newaxis, 1:])
+    lower = np.maximum(new_breaks[:-1, np.newaxis],
+                       old_breaks[np.newaxis, :-1])
+
+    weights = np.maximum(upper - lower, 0)
+    weights /= weights.sum(axis=1, keepdims=True)
+
+    return weights
+
+
+def resize_local_mean(image, output_shape, grid_mode=True,
+                      preserve_range=False, *, channel_axis=None):
+    """Resize an array with the local mean / bilinear scaling.
+
+    Parameters
+    ----------
+    image : ndarray
+        Input image. If this is a multichannel image, the axis corresponding
+        to channels should be specified using `channel_axis`
+    output_shape : tuple or ndarray
+        Size of the generated output image. When `channel_axis` is not None,
+        the `channel_axis` should either be omitted from `output_shape` or the
+        ``output_shape[channel_axis]`` must match
+        ``image.shape[channel_axis]``. If the length of `output_shape` exceeds
+        image.ndim, additional singleton dimensions will be appended to the
+        input ``image`` as needed.
+    grid_mode : bool, optional
+        If False, the distance from the pixel centers is zoomed. Otherwise, the
```

Comment:
This sentence is not clear to me. It is specific to `scipy.ndimage.zoom`.
I like the example and the illustration, but the description is confusing to me :confused:.
What about
```
Defines ``image`` pixels position: if True, pixels are assumed to be at grid intersections,
otherwise at cell centers. As a consequence, for example, a 1d ...
```

Reply:
done